app_flutter/lib/task_box.dart

```diff
@@ -215,38 +212,119 @@ class TaskOverlayContents extends StatelessWidget {
           // Move this overlay to be where the parent is
           top: offsetLeft.dy + (renderBox.size.height / 2),
           left: offsetLeft.dx + (renderBox.size.width / 2),
-          child: Card(
-            child: Column(
-              mainAxisSize: MainAxisSize.max,
-              children: <Widget>[
-                ListTile(
-                  leading: Tooltip(
-                      message: taskStatus, child: statusIcon[taskStatus]),
-                  title: Text(task.name),
-                  subtitle: Text(
-                      'Attempts: ${task.attempts}\nDuration: ${task.endTimestamp - task.startTimestamp} seconds\nAgent: ${task.reservedForAgentId}'),
-                ),
-                ButtonBar(
-                  children: <Widget>[
-                    IconButton(
-                      icon: const Icon(Icons.receipt),
-                      onPressed: () {
-                        // TODO(chillers): Open log in new window. https://github.com/flutter/cocoon/issues/436
-                      },
-                    ),
-                    IconButton(
-                      icon: const Icon(Icons.redo),
-                      onPressed: () {
-                        // TODO(chillers): Rerun task. https://github.com/flutter/cocoon/issues/424
-                      },
-                    ),
-                  ],
-                ),
-              ],
-            ),
+          child: TaskOverlayContents(
+            parentContext: parentContext,
+            buildState: buildState,
+            task: task,
+            taskStatus: taskStatus,
           ),
         ),
       ],
     );
   }
 }
+
+/// Displays the information from [Task] and allows interacting with a [Task].
+///
+/// This is intended to be inserted in [TaskOverlayEntry].
+///
+/// Offers the functionality of opening the log for this [Task] and rerunning
+/// this [Task] through the build system.
+class TaskOverlayContents extends StatelessWidget {
+  const TaskOverlayContents({
+    Key key,
+    @required this.parentContext,
+    @required this.buildState,
+    @required this.task,
+    @required this.taskStatus,
+  })  : assert(parentContext != null),
+        assert(buildState != null),
+        assert(task != null),
+        super(key: key);
+
+  /// The parent context that has the size of the whole screen
+  final BuildContext parentContext;
```

Comment:
It looks like you're only using this to show a snackbar, can you use the child context instead?

Reply:
I tried that initially and the `SnackBar` wouldn't appear, but using the parent context worked.

Reply:
Oh right, this is in an overlay, so it doesn't actually have access to the context with a Scaffold parent. In this case, I would instead pass down a callback that takes a Snackbar and internally invokes parentContext.showSnackbar - this cuts down the API surface/coupling dramatically.

Reply:
Like
`  final ScaffoldFeatureController<SnackBar, SnackBarClosedReason> Function(
      SnackBar) showSnackbarCallback;`?